packages/react-core/src/components/Modal/Modal.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import ModalContent from './ModalContent';
+
+const propTypes = {
+  /** content rendered inside the Modal. */
+  children: PropTypes.node.isRequired,
+  /** additional classes added to the button */
+  className: PropTypes.string,
+  /** Flag to show the modal */
+  show: PropTypes.bool,
+  /** Content of the Modal Header */
+  header: PropTypes.any,
+  /** Content of the Modal Footer */
+  footer: PropTypes.any,
+  /** A callback for when the close button is clicked */
+  onClose: PropTypes.func,
+  /** Creates a large version of the Modal */
+  isLarge: PropTypes.bool,
+  /* Aria label used for Modal Box Header */
+  label: PropTypes.string.isRequired
+};
+
+const defaultProps = {
+  className: '',
+  show: false,
+  header: null,
+  footer: null,
+  onClose: () => undefined,
+  isLarge: false
+};
+
+// create a node for the modal and render it to the DOM
+const modalRoot = document.createElement('div');
+modalRoot.setAttribute('id', 'modal-root');
+document.body.appendChild(modalRoot);
+
+class Modal extends React.Component {
+  static propTypes = propTypes;
+  static defaultProps = defaultProps;
+
+  constructor(props) {
+    super(props);
+    this.container = document.createElement('div');
+  }
+
+  componentDidMount() {
+    modalRoot.appendChild(this.container);
+  }
+
+  componentWillUnmount() {
+    modalRoot.removeChild(this.container);
+  }
+
+  render() {
+    return ReactDOM.createPortal(
```

Comment:
Please don't display Modals over ReactDom, that's quite a bad way to do it when we want to have small components without any sidefects. You can just simply pass `{isShown && <ModalContent {...props} />}` instead of this magic. This overcomplicated component can be rewritten in this way. 
```jsx
import React from 'react';
import PropTypes from 'prop-types';
import ModalContent from './ModalContent';

const propTypes = {
  isOpen: PropTypes.bool
};

const defaultProps = {
  isOpen: false
};
const Modal = ({ isOpen, ...props }) => (
  <React.Fragment>
    {isOpen && <ModalContent {...props} /> }
  </React.Fragment>
);
Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;
```

Reply:
@karelhala do we usually have the consumer provide an ID via props?

Reply:
no.  We will generate it and pass it down to the components.  at its simplest form it could just be 
```jsx
let currentId = 0;
class Modal extends React.Component {
  id = `pf-modal-${currentId++}`;
  
  render() { .
      return ReactDOM.createPortal(<ModalContent id={this.id} />, this.target);
   }
}
```

Reply:
Yeah I understand, that there might be problem with `overflow: hidden` in parent or something like that, but I still think that the DOM placement should be done by consumer rather then by us. So any application consuming this modal would decide where to store and if they want to use `createPortal` and in which phase.

I guess that the best way of doing this would be to provide both ways, `Modal` component which will use predefined portal placement and `ModalContent` as well. So consumer of this repository has choice where to place the Modal.

Plus it would be really great if we'd allow consumers to define their DOM element where to place this Modal which would default to `body`.

Reply:
I am good with those changes.  I think we should provide the defaults we have in here, and then allow the customizations you mentioned.  I really like how Atlaskit does it with their LayerManager so that you can render multiple Layers.  It is pretty complicated though so it might be that we want to add something like that in a later PR.  For now we can offer a `target`/`container` prop, don't care about the name.  That allows users to specify their own.  This will match how material-ui does it which is probably the most popular react component lib.